packages/cli/src/build/__tests__/src-root.ts

```diff
@@ -0,0 +1,83 @@
+import build from "../";
+import { testdir, getDist, ts, repoNodeModules } from "../../../test-utils";
+
+test("monorepo with root src dir and typescript package", async () => {
+  let tmpPath = await testdir({
+    "package.json": JSON.stringify({
+      name: "root",
+      devDependencies: {
+        typescript: "^3.8.3",
+      },
+      preconstruct: {
+        packages: ["src/packages/*"],
+      },
+    }),
+    ".babelrc": JSON.stringify({
+      presets: [
+        require.resolve("@babel/preset-env"),
+        require.resolve("@babel/preset-typescript"),
+      ],
+    }),
+    node_modules: {
+      kind: "symlink",
+      path: repoNodeModules,
+    },
+    "src/packages/my-package/package.json": JSON.stringify({
+      name: "my-package",
+      main: "dist/my-package.cjs.js",
+      module: "dist/my-package.esm.js",
+    }),
+    "src/packages/my-package/tsconfig.json": `{
+    "compilerOptions": {
+      "target": "esnext",
+      "module": "commonjs",
+      "outDir": "./build"
+      "isolatedModules": true,
+      "esModuleInterop": true,
+      "noEmit": true
+    }
+  }
+  `,
+
+    "src/packages/my-package/src/index.ts": ts`
+                                              export default {};
+                                            `,
+  });
+
+  await build(tmpPath);
+
+  expect(await getDist(tmpPath + "/src/packages/my-package"))
+    .toMatchInlineSnapshot(`
+  ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ dist/declarations/build/index.d.ts ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
```

Comment:
The fact that `outDir` is respected is quite bad imo since this doesn't really do what you might expect (and it would still break if for example you had a tsconfig in your project root with `outDir`) so I'd rather just ignore the option. Could you change this to delete `outDir` from the parsed config like how `declaration`, `emitDeclarationOnly` and `noEmit` are set? (Ideally, preconstruct would probably error if `outDir` is set but that would potentially be breaking so let's avoid that for now)

Reply:
Made the change as requested, also unset `declarationDir` because that has a similar effect.